test/controllers/provisioning_templates_controller_test.rb

```diff
@@ -144,6 +144,18 @@ class ProvisioningTemplatesControllerTest < ActionController::TestCase
       get :build_pxe_default, {}, set_session_user
       assert_redirected_to provisioning_templates_path
     end
+
+    test "pxe menu's labels should be sorted by full hostname title" do
+      t1 = TemplateCombination.new :hostgroup => hostgroups(:inherited), :environment => environments(:production)
+      t1.provisioning_template = templates(:mystring2)
+      t2 = TemplateCombination.new :hostgroup => hostgroups(:nested), :environment => environments(:production)
+      t2.provisioning_template = templates(:mystring2)
+      t1.save
```

Comment:
Use `TemplateCombination.create` to avoid all of this editing and saving, or preferably just use the :template_combination factory.